libr/core/cmd_print.c

```diff
@@ -5854,97 +5854,8 @@ l = use_blocksize;
 			break;
 		case 't': // "pst"
 			if (l > 0) {
-				/*SdbListIter *iter;
-				SdbKv *kv;
-				SdbList *sdbls = sdb_foreach_list (core->anal->sdb_meta, true);
-				ut8 out[len];
-				ut8 c;
-				int i = 0;
-
-				for (int j = 0; j < len; j ++) {
-					ls_foreach (sdbls, iter, kv) {
-							char hex_str[2];
-							memcpy(hex_str, sdbkv_key (kv)+2, sizeof(char)*2);
-							int hex = (int)strtol(hex_str, NULL, 16);
-							if (core->block[j] == hex) {
-									c = (ut8) sdbkv_value (kv)[0];
-									out[i] = c;
-									i++;
-							}
-							out[i] = '\0';
-						}
-					}*/
-
-				charset pokered[61] = {
-											{ '\0', 0x50 },//terminating byte
-
-										    { ' ', 0x7f },
-										    { 'A', 0x80 },
-										    { 'B', 0x81 },
-										    { 'C', 0x82 },
-										    { 'D', 0x83 },
-										    { 'E', 0x84 },
-										    { 'F', 0x85 },
-										    { 'G', 0x86 },
-										    { 'H', 0x87 },
-										    { 'I', 0x88 },
-										    { 'J', 0x89 },
-										    { 'K', 0x8a },
-										    { 'L', 0x8b },
-										    { 'M', 0x8c },
-										    { 'N', 0x8d },
-										    { 'O', 0x8e },
-										    { 'P', 0x8f },
-										    { 'Q', 0x90 },
-										    { 'R', 0x91 },
-										    { 'S', 0x92 },
-										    { 'T', 0x93 },
-										    { 'U', 0x94 },
-										    { 'V', 0x95 },
-										    { 'W', 0x96 },
-										    { 'X', 0x97 },
-										    { 'Y', 0x98 },
-										    { 'Z', 0x99 },
-
-
-										    { '(', 0x9a },
-										    { ')', 0x9b },
-										    { ':', 0x9c },
-										    { ';', 0x9d },
-										    { '[', 0x9e },
-										    { ']', 0x9f },
-										    { 'a', 0xa0 },
-										    { 'b', 0xa1 },
-										    { 'c', 0xa2 },
-										    { 'd', 0xa3 },
-										    { 'e', 0xa4 },
-										    { 'f', 0xa5 },
-										    { 'g', 0xa6 },
-										    { 'h', 0xa7 },
-										    { 'i', 0xa8 },
-										    { 'j', 0xa9 },
-										    { 'k', 0xaa },
-										    { 'l', 0xab },
-										    { 'm', 0xac },
-										    { 'n', 0xad },
-										    { 'o', 0xae },
-										    { 'p', 0xaf },
-										    { 'q', 0xb0 },
-										    { 'r', 0xb1 },
-										    { 's', 0xb2 },
-										    { 't', 0xb3 },
-										    { 'u', 0xb4 },
-										    { 'v', 0xb5 },
-										    { 'w', 0xb6 },
-										    { 'x', 0xb7 },
-										    { 'y', 0xb8 },
-										    { 'z', 0xb9 },
-
-											{ '-', 0xe3 },
-
-								   };
 				ut8 var[len];
-				r_charset_encode_str (var, core->block, len, pokered, 61);
+				r_charset_encode_str (var, core->block, len, custom_charset, 61);
```

Comment:
I will handle the case that `pse` has not been typed yet.

Reply:
why 61

Reply:
@radare it is the size of the charset encoding.

Reply:
@radare it is the size of the custom charset.

Reply:
maybe a #define would make that nicer instead of a value that suddenly appears

Reply:
@Maijin It is temporary. I am going to try with a function similar to strlen() to find the size of the struct array custom_charset. 61 is temporary.

Do you think that the use a preprocessor is a good idea? This implies that each custom charset plugin has the same size. I do not think it should be the case.

Reply:
Ok in this case should be different indeed. I suggested preprocessor in case you had only one entry to avoid magic values.

Reply:
@Maijin I think I can get the i here https://github.com/radareorg/radare2/pull/16621/files#diff-55abf099b2aa9100745103d790a1ad9cR5921 in a global-variable-like in c in the header file.

Reply:
@Maijin edit: I will make a self referenced structure instead of typedef struct { ut8 ch; ut8 hx; } charset[61];